contracts/WooRebateManager.sol

```diff
@@ -0,0 +1,131 @@
+    using SafeMath for uint256;
+    using DecimalMath for uint256;
+    using SafeERC20 for IERC20;
+
+    mapping(address => uint256) public override rebateRate; // decimal: 18; 1e16 = 1%, 1e15 = 0.1%, 1e16 = 0.01%
+    mapping(address => uint256) public pendingRebate; // pending rebate amount in quote token
+
+    IWooPP private wooPP;
+
+    address public immutable quoteToken; // USDT
+    address public immutable rewardToken; // WOO
+
+    constructor(address newQuoteToken, address newRewardToken) public {
+        require(newQuoteToken != address(0), 'WooRebateManager: INVALID_QUOTE');
+        require(newRewardToken != address(0), 'WooRebateManager: INVALID_RAWARD_TOKEN');
+        initOwner(msg.sender);
+        quoteToken = newQuoteToken;
+        rewardToken = newRewardToken;
+    }
+
+    function addRebate(address brokerAddr, uint256 amountInUSD) external override {
+        if (brokerAddr == address(0)) {
+            return;
+        }
+
+        uint256 balanceBefore = IERC20(quoteToken).balanceOf(address(this));
+        TransferHelper.safeTransferFrom(quoteToken, msg.sender, address(this), amountInUSD);
+        uint256 balanceAfter = IERC20(quoteToken).balanceOf(address(this));
+        require(balanceAfter.sub(balanceBefore) >= amountInUSD, 'RebateManager: amount < balance delta');
+
+        pendingRebate[brokerAddr] = amountInUSD.add(pendingRebate[brokerAddr]);
+    }
+
+    function pendingRebateInUSDT(address brokerAddr) external view override returns (uint256) {
+        require(brokerAddr != address(0), 'WooRebateManager: zero_brokerAddr');
+        return pendingRebate[brokerAddr];
+    }
+
+    function pendingRebateInWoo(address brokerAddr) external view override returns (uint256) {
+        require(brokerAddr != address(0), 'WooRebateManager: zero_brokerAddr');
+        return wooPP.querySellQuote(rewardToken, pendingRebate[brokerAddr]);
+    }
+
+    function claimRebate() external override {
+        require(pendingRebate[msg.sender] > 0, 'WooRebateManager: NO_pending_rebate');
+
+        uint256 quoteAmount = pendingRebate[msg.sender];
+        uint256 balanceBefore = IERC20(rewardToken).balanceOf(address(this));
+        TransferHelper.safeApprove(quoteToken, address(wooPP), quoteAmount);
+        uint256 wooAmount = IWooPP(wooPP).sellQuote(rewardToken, quoteAmount, 0, address(this), address(0));
+        uint256 balanceAfter = IERC20(rewardToken).balanceOf(address(this));
+        require(balanceAfter.sub(balanceBefore) >= wooAmount, 'WooRebateManager: woo amount INSUFF');
+
+        TransferHelper.safeTransfer(rewardToken, msg.sender, wooAmount);
+        pendingRebate[msg.sender] = 0;
+        emit ClaimReward(msg.sender, wooAmount);
+    }
+
+    function setRebateRate(address brokerAddr, uint256 rate) external override onlyOwner {
+        require(brokerAddr != address(0), 'WooRebateManager: brokerAddr_ZERO_ADDR');
+        require(rate <= 1e16, 'WooRebateManager: INVALID_USER_REWARD_RATE'); // rate <= 1%
```

Comment:
`rebateAmount = rate * feeAmount`, right? So we should expect rebate rate to be higher than 1%.

Reply:
should be  1e18 (100%)